sdk/src/transaction.rs

```diff
@@ -255,6 +255,62 @@ impl Transaction {
         }
     }
 
+    /// Check keys and keypair lengths, then sign this transaction, returning any signing errors
+    /// encountered
+    pub fn try_sign<T: Signers>(
```

Comment:
Is this all copy-pasta?  Any way we can have the old functions all these new ones and then panic if there's an error?

Reply:
Totally